dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+        }
+
+        Collections.sort(roles);
+
+
+        for (Role role : roles) {
+            HibernateUtil _dh = new HibernateUtil(Role.class);
+            String id = HibernateUtil.getSession().getSessionFactory().getClassMetadata(Role.class).getIdentifierPropertyName();
+
+            if(UtilMethods.isSet(role.getRoleKey())) {
+               List<Map<String,Object>>  matches= new DotConnect().setSQL("select * from cms_role where role_key =?").addParam(role.getRoleKey()).loadObjectResults();
+               Logger.info(this.getClass(), "roleKey:" +role.getRoleKey() + " = " + matches.size() );
+            }
+
+            _dh.saveWithPrimaryKey(role, role.getId());
+            HibernateUtil.getSession().flush();
+        }
+
+        for (File file : endsWith(User.class.getCanonicalName() + ".xml")) {
+            doXMLFileImport(file);
+        }
+
+        for (File file : contains(UsersRoles.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains(Layout.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+
+        
+        
+        for (File file : contains(PortletsLayouts.class.getCanonicalName())) {
+            doXMLFileImport(file);
+        }
+
+        
+        for (File file : endsWith("LogsMappers.xml")) {
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains(LayoutsRoles.class.getCanonicalName())) {
+            doXMLFileImport(file);
+        }
+        
+        
+        
+        for (File file : contains(Language.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+        /*
+         * Because of the parent check we do at db we need to import folder identifiers first but for the
+         * same reason we need to sort them first by parent_path
+         */
+        final List<Identifier> folderIdents = new ArrayList<Identifier>();
+        final List<Identifier> otherIdents = new ArrayList<Identifier>();
+
+        List<File> identifiers = contains(Identifier.class.getCanonicalName());
+        // collecting all folder identifiers
+        for (File ff : identifiers) {
+            List<Identifier> idents = (List<Identifier>) BundlerUtil.xmlToObject(ff);
+            for (Identifier ident : idents) {
+                if (ident.getAssetType().equals("folder")) {
+                    folderIdents.add(ident);
+                }else {
+                    otherIdents.add(ident);
+                }
+            }
+        }
+
+        // sorting folder identifiers by parent path in order to pass parent check
+        Collections.sort(folderIdents, new Comparator<Identifier>() {
+            public int compare(Identifier o1, Identifier o2) {
+                return o1.getParentPath().compareTo(o2.getParentPath());
+            }
+        });
+
+        // saving folder identifiers
+        for (Identifier ident : folderIdents) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'ident' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)